src/uicommon.ml

```diff
@@ -493,17 +493,23 @@ let calcETA rem rate =
   let t = truncate (rem /. rate +. 0.5) in
   (* Estimating the remaining time is not accurate. Reduce the display
      precision (and reduce more when longer time remaining). *)
-  let h, (m, sec) =
-    if t >= 3420 then
-      let u = t + 180 in u / 3600, (((u mod 3600) / 300) * 5, 0)
-    else
-      0,
-      if t >= 2640 then ((t + 180) / 300) * 5, 0
-      else if t >= 1800 then ((t + 119) / 120) * 2, 0
-      else if t >= 120 then let u = t + 15 in u / 60, ((u mod 60) / 30) * 30
-      else t / 60, t mod 60
-  in
-  Printf.sprintf "%02d:%02d:%02d" h m sec
+  if t >= 86220 then
+    let d, h, (m, sec) =
+      let u = t + 180 in u / 86400, (u mod 86400) / 3600, (((u mod 3600) / 300) * 5, 0)
+    in
+    Printf.sprintf "%dd %02d:%02d:%02d" d h m sec
```

Comment:
Overall looks correct but I propose to simplify a little since now you don't need the tuple at all. At the same time, I'd consider reducing the precision even more.
Something like this (not tested):
```suggestion
    let u = t + 1799 in
    Printf.sprintf "%dd %02d:%02d:00" (u / 86400) ((u mod 86400) / 3600)
      (((u mod 3600) / 1800) * 30)
```
This makes the precision 30 min (currently rounded fully up, but can be adjusted by changing the constant `1799`). This could be anything, really.

Alternatively, you can still express all variables separately, but still don't need the tuple:
```ocaml
    let u = t + 1799 in
    let d = u / 86400 in
    let h = (u mod 86400) / 3600 in
    let m = ((u mod 3600) / 1800) * 30 in
    Printf.sprintf "%dd %02d:%02d:00" d h m
```
but I'm not sure it adds anything.

Reply:
Fine with me, but then I would `let u = t + 1800` and make the threshold 84800 (86400 - 1800), to be consistent with the rest of the code.

Reply:
You could, but is it the best idea? Those thresholds are not directly related. You can still keep it at 86220 because that goes together with the < 24h threshold. If you make the threshold (86400 - 1800) then you cut the 24h range a bit short (meaning you'll jump from 23:30:00 to 1d 00:00:00). But that's ok too, all of this is completely arbitrary; there is nothing specifying that it should be 5 min precision up to 24h.

Reply:
> to be consistent with the rest of the code.

That's not making it consistent. Those thresholds are not related and you can see these differ in the code below. They are arbitrary, true, but they are not the same.

Reply:
Indeed, I just went by the next lower threshold: for ETAs above roughly an hour, we add 180 s and make the threshold 3420 s.

Looking at the code again, however, I realize that for any `3420 > t >= 2640`, we effectively do the same as for `86400-ish > t >= 3420`, so we could simplify that part as well.

And below that, the thresholds are “true” rather than rounded ETA.

Reconsidering it all, I’d lean towards sticking with 5-minute precision up to `1d 00:00:00` or `24:00:00` (I guess the former format would be more consistent). So that would put us back to 86220...

Reply:
In short, no further commits on top of your suggested improvements. The rest of the commit history in this PR cancels itself out.